pkg/agent/ipassigner/ip_assigner_linux.go

```diff
@@ -41,29 +40,45 @@ type ipAssigner struct {
 	// assignIPs caches the IPs that are assigned to the dummy device.
 	// TODO: Add a goroutine to ensure that the cache is in sync with the IPs assigned to the dummy device in case the
 	// IPs are removed by users accidentally.
-	assignedIPs sets.String
-	mutex       sync.RWMutex
+	assignedIPs  sets.String
+	mutex        sync.RWMutex
+	arpResponder responder.Responder
+	ndpResponder responder.Responder
 }
 
 // NewIPAssigner returns an *ipAssigner.
 func NewIPAssigner(nodeTransportInterface string, dummyDeviceName string) (*ipAssigner, error) {
-	_, _, externalInterface, err := util.GetIPNetDeviceByName(nodeTransportInterface)
+	ipv4, ipv6, externalInterface, err := util.GetIPNetDeviceByName(nodeTransportInterface)
 	if err != nil {
 		return nil, fmt.Errorf("get IPNetDevice from name %s error: %+v", nodeTransportInterface, err)
 	}
-
-	dummyDevice, err := ensureDummyDevice(dummyDeviceName)
-	if err != nil {
-		return nil, fmt.Errorf("error when ensuring dummy device exist: %v", err)
-	}
-
 	a := &ipAssigner{
 		externalInterface: externalInterface,
-		dummyDevice:       dummyDevice,
 		assignedIPs:       sets.NewString(),
 	}
-	if err := a.loadIPAddresses(); err != nil {
-		return nil, fmt.Errorf("error when loading IP addresses from the system: %v", err)
+	if ipv4 != nil {
+		arpResonder, err := responder.NewARPResponder(externalInterface)
```

Comment:
As ARP is not always needed, can we pass a boolean to enable ARP responder? We can add another for NDP too.

Or another we do not enable ARP when dummyDeviceName is not "".

Reply:
updated.

Reply:
I do not see you changed this?